src/containers/my-resources/questions-container/QuestionsContainer.tsx

```diff
@@ -52,106 +46,105 @@ const QuestionsContainer = () => {
     removeColumnRules
   )
 
-  const onResponseError = useCallback(
-    (error?: ErrorResponse) => {
-      dispatch(
-        openAlert({
-          severity: snackbarVariants.error,
-          message: getErrorKey(error)
-        })
-      )
-    },
-    [dispatch]
-  )
-
-  const getQuestions = useCallback(
-    () =>
-      ResourceService.getQuestions({
-        limit: itemsPerPage,
-        sort,
-        title: searchTitle.current,
-        skip: (page - 1) * itemsPerPage,
-        categories: selectedItems
-      }),
-    [itemsPerPage, sort, page, selectedItems]
-  )
-
-  const { response, loading, fetchData } = useAxios<
-    ItemsWithCount<Question>,
-    GetResourcesParams
-  >({
-    service: getQuestions,
-    defaultResponse: defaultResponses.itemsWithCount,
-    onResponseError
+  const getQuestions = useCallback(() => {
+    return ResourceService.getQuestionsQuery({
+      limit: itemsPerPage,
+      sort,
+      title: searchTitle.current,
+      skip: (page - 1) * itemsPerPage,
+      categories: selectedItems
+    })
+  }, [itemsPerPage, sort, page, selectedItems])
+
+  const {
+    data: questions,
+    isLoading: isQuestionsLoading,
+    error: questionsError,
+    refetch: refetchQuestions
+  } = useQuery({
+    queryKey: ['questions', itemsPerPage, sort, page, selectedItems],
+    queryFn: getQuestions,
+    options: {
+      staleTime: Infinity
+    }
   })
 
   const deleteQuestion = useCallback(
-    (id?: string) => ResourceService.deleteQuestion(id ?? ''),
-    []
+    async (id?: string): Promise<AxiosResponse<unknown>> => {
+      const response = await ResourceService.deleteQuestion(id ?? '')
+      await queryClient.invalidateQueries({ queryKey: ['questions'] })
+      return response
+    },
+    [queryClient]
   )
 
   const editQuestion = (id: string) => {
-    navigate(createUrlPath(authRoutes.myResources.editQuestion.path, id))
+    return navigate(
```

Comment:
You don't need `navigate` return value
```suggestion
    navigate(
```